pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -279,66 +280,75 @@ const SourceRow = ({ source, sourceType, networks, nodeDevices, providerName, on
     );
 };
 
-const OSRow = ({ vendor, osInfoList, os, vendors, onValueChanged, validationFailed }) => {
-    const validationStateOsVendor = validationFailed.vendor ? 'error' : undefined;
-    const familyList = vendors.familyList;
-    const familyMap = vendors.familyMap;
-    const vendorMap = vendors.vendorMap;
-    const vendorSelectEntries = [];
-
-    if (familyMap[DIVIDER_FAMILY]) {
-        vendorSelectEntries.push((
-            <Select.SelectEntry data={NOT_SPECIFIED} key={NOT_SPECIFIED}>{_(NOT_SPECIFIED)}</Select.SelectEntry>));
-        vendorSelectEntries.push((<Select.SelectDivider key='divider' />));
-    }
+class OSRow extends React.Component {
+    constructor(props) {
+        super(props);
+        const IGNORE_VENDORS = ['ALTLinux', 'Mandriva', 'GNOME Project'];
+        const osInfoListExt = this.props.osInfoList
+                .filter(os => !IGNORE_VENDORS.find(vendor => vendor == os.vendor))
+                .map(os => correctSpecialCases(os))
+                .filter(os => filterReleaseEolDates(os))
+                .sort((a, b) => {
+                    if (a.vendor == b.vendor)
+                        if (a.releaseDate || b.releaseDate)
+                            return compareDates(a.releaseDate, b.releaseDate, true) > 0;
+                        else
+                            return a.version < b.version;
+                    else
+                        return getOSStringRepresentation(a).toLowerCase() > getOSStringRepresentation(b).toLowerCase();
+                });
 
-    familyList.forEach(({ family, vendors }) => {
-        if (family === DIVIDER_FAMILY) {
-            return;
-        }
+        osInfoListExt.push({
+            'shortId': OTHER_OS_SHORT_ID,
+            'name': OTHER_OS,
+            'version': null,
+        });
 
-        vendorSelectEntries.push(
-            <optgroup key={family} label={family}>
-                { vendors.map(vendor => <Select.SelectEntry data={vendor} key={vendor}>{vendor}</Select.SelectEntry>) }
-            </optgroup>);
-    });
-
-    const osEntries = (
-        vendorMap[vendor]
-                .map(os => (
-                    <Select.SelectEntry data={os.shortId} key={os.shortId}>
-                        {getOSStringRepresentation(os)}
-                    </Select.SelectEntry>))
-    );
+        this.state = {
+            typeAheadKey: Math.random(),
```

Comment:
Using random as key is not wrong but not optimal. Is every item going to be in the list just once, if so, then you can construct some string like `shortId+name+version`.

Reply:
This is done on the purpose actually. See onBlur callback on the TypeAheadSelect.

The thing is that, by default that component, when you onfocus, it will keep in the input text whatever you typed, regardless it's matching with one of the preselected elemements.
I couldn't set the value of the input field, (this make the component controlled)
So I decided to reset the whole component when onBlur event arrives.
The only clear way to reset a react component, it to change it's key.
I will be glad it you have some suggestion on how to ger rid of this ugly hack of course!

Reply:
> Small nitpicks.
> Also there is typo in one commit title `machines: OS now are filtered by shortId as well the OS diplay name` and one of @garrett commit could be squashed.
> 
> And I think that you know, but tests still don't seems to get green :(

Test issue hopefully fixed with repost.